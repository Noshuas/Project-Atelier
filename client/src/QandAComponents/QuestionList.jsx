import React from 'react';
import Question from './Question.jsx';

const QuestionList = (props) => {
  return (
    <div>
      < Question />
      < Question />
      < Question />
      < Question />
    </div>
  );
};

export default QuestionList;