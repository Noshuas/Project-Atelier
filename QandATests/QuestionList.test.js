/**
 * @jest-environment jsdom
*/
import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import QuestionList from '../client/src/QandAComponents/QuestionList.jsx';
import ReactDOM from 'react-dom';
import Helpers from '../client/src/QandAComponents/helpers.js';
import { render, screen, cleanup } from '@testing-library/react';

//after
const testResults = [{
  'question_id': 1,
  'question_body': 'Test 1?',
  'question_date': '2018-10-18T00:00:00.000Z',
  'asker_name': 'williamsmith',
  'question_helpfulness': 10,
  'reported': false,
  'answers': {
    68: {
      'id': 68,
      'body': 'We are selling it here without any markup from the middleman!',
      'date': '2018-08-18T00:00:00.000Z',
      'answerer_name': 'Seller',
      'helpfulness': 4,
      'photos': []
      // ...
    }
  }
},
{
  'question_id': 2,
  'question_body': 'Test 2?',
  'question_date': '2018-10-18T00:00:00.000Z',
  'asker_name': 'williamsmith',
  'question_helpfulness': 10,
  'reported': false,
  'answers': {
    68: {
      'id': 68,
      'body': 'We are selling it here without any markup from the middleman!',
      'date': '2018-08-18T00:00:00.000Z',
      'answerer_name': 'Seller',
      'helpfulness': 4,
      'photos': []
      // ...
    }
  }
},
{
  'question_id': 3,
  'question_body': 'Should not display unless expanded',
  'question_date': '2018-10-18T00:00:00.000Z',
  'asker_name': 'williamsmith',
  'question_helpfulness': 10,
  'reported': false,
  'answers': {
    68: {
      'id': 68,
      'body': 'We are selling it here without any markup from the middleman!',
      'date': '2018-08-18T00:00:00.000Z',
      'answerer_name': 'Seller',
      'helpfulness': 4,
      'photos': []
      // ...
    }
  }
}];
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<QuestionList questions={testResults} expanded={false} productName='Test Product' />, div);
});

test('QuestionList snapshot 1', () => {
  const component = renderer.create(
    <QuestionList questions={[]} expanded={false} productName='Test Product' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('QuestionList snapshot 2', () => {
  const component = renderer.create(
    <QuestionList questions={testResults} expanded={false} productName='Test Product' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
