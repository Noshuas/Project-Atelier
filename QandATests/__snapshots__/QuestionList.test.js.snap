// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionList snapshot 1 1`] = `
<div
  className="question-list"
  data-testid="question-list"
/>
`;

exports[`QuestionList snapshot 2 1`] = `
<div
  className="question-list"
  data-testid="question-list"
>
  <div
    className="question-section"
  >
    <div
      className="question-main"
    >
      <h3>
        Q: 
        Test 1?
         
      </h3>
      <div
        className="question-toolbar"
      >
        <span>
          <span>
             Helpful? 
            <a
              href="#"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            10
            )
          </span>
           | 
          <a
            href="#"
            onClick={[Function]}
          >
            Add Answer
          </a>
        </span>
      </div>
    </div>
    <h3>
      A:
    </h3>
    <div>
      <span>
        This question currently has no answers
      </span>
    </div>
  </div>
  <div
    className="question-section"
  >
    <div
      className="question-main"
    >
      <h3>
        Q: 
        Test 2?
         
      </h3>
      <div
        className="question-toolbar"
      >
        <span>
          <span>
             Helpful? 
            <a
              href="#"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            10
            )
          </span>
           | 
          <a
            href="#"
            onClick={[Function]}
          >
            Add Answer
          </a>
        </span>
      </div>
    </div>
    <h3>
      A:
    </h3>
    <div>
      <span>
        This question currently has no answers
      </span>
    </div>
  </div>
</div>
`;

exports[`QuestionList snapshot 3 1`] = `
<div
  className="question-list"
  data-testid="question-list"
>
  <div
    className="question-section"
  >
    <div
      className="question-main"
    >
      <h3>
        Q: 
        Test 1?
         
      </h3>
      <div
        className="question-toolbar"
      >
        <span>
          <span>
             Helpful? 
            <a
              href="#"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            10
            )
          </span>
           | 
          <a
            href="#"
            onClick={[Function]}
          >
            Add Answer
          </a>
        </span>
      </div>
    </div>
    <h3>
      A:
    </h3>
    <div>
      <span>
        This question currently has no answers
      </span>
    </div>
  </div>
  <div
    className="question-section"
  >
    <div
      className="question-main"
    >
      <h3>
        Q: 
        Test 2?
         
      </h3>
      <div
        className="question-toolbar"
      >
        <span>
          <span>
             Helpful? 
            <a
              href="#"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            10
            )
          </span>
           | 
          <a
            href="#"
            onClick={[Function]}
          >
            Add Answer
          </a>
        </span>
      </div>
    </div>
    <h3>
      A:
    </h3>
    <div>
      <span>
        This question currently has no answers
      </span>
    </div>
  </div>
  <div
    className="question-section"
  >
    <div
      className="question-main"
    >
      <h3>
        Q: 
        Should not display unless expanded
         
      </h3>
      <div
        className="question-toolbar"
      >
        <span>
          <span>
             Helpful? 
            <a
              href="#"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            10
            )
          </span>
           | 
          <a
            href="#"
            onClick={[Function]}
          >
            Add Answer
          </a>
        </span>
      </div>
    </div>
    <h3>
      A:
    </h3>
    <div>
      <span>
        This question currently has no answers
      </span>
    </div>
  </div>
</div>
`;
